Await the adapt call of a dask cluster instance if it's an asynchronous method

john-jam opened the pull request:
<!-- Thanks for contributing 🎉! Please ensure the title neatly summarizes the proposed changes. -->

<!-- Overview -->
This PR will await the `adapt` call when the underlying dask cluster implementation returns a future for that method.

As discussed in the [issue](https://github.com/PrefectHQ/prefect-dask/issues/57), it seemed like an update that should be done upstream in the `dask-kubernetes` package but [according to its maintainer](https://github.com/dask/dask-kubernetes/issues/645#issuecomment-1403426252), the new `operator.KubeCluster` is doing I/O during adapt so it should in theory be awaited when run asynchronously.

Closes #57 

### Checklist
<!-- These boxes may be checked after opening the pull request. -->

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-dask/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-dask/blob/main/CHANGELOG.md)

ahuang11 commented:
Thanks for contributing this! Would you be able to help add a CHANGELOG entry and also a test with a mock cluster that has an awaitable   adapt method?

john-jam commented:
@ahuang11 I added a test to check if the `adapt` method was called properly and updated the CHANGELOG as requested. Since the cluster class instantiation happens inside the DaskTaskRunner class, I am not sure how to use a unittest.Mock instance in that case so I just added dummy Mocks in the test itself. Please let me know if there's a more elegant way of doing this.
